src/native/ckmeans.c

```diff
@@ -0,0 +1,184 @@
+#include <stdlib.h>
+#include <math.h>
+
+
+// nClusters columns nValues rows
+int compare_doubles(const void *p, const void *q) {
+    double x = *(const double *)p;
+    double y = *(const double *)q;
+    if (x < y) {
+        return -1;
+    } else if (x > y) {
+        return 1;
+    }
+    return 0;
+}
+
+size_t unique_sort_count (double array[], size_t len) {
+    if (len == 0) {
+        return 0;
+    }
+    size_t uniqueValueCount = 1;
+    double lastSeenValue = array[0];
+    for (size_t i = 1; i < len; i++) {
+        if (array[i] != lastSeenValue) {
+            uniqueValueCount++;
+            lastSeenValue = array[i];
+        }
+    }
+    return uniqueValueCount;
+}
+size_t min_t (size_t a, size_t b) {
+    if (a < b) {
+        return a;
+    }
+    return b;
+}
+size_t max_t (size_t a, size_t b) {
+    if (a > b) {
+        return a;
+    }
+    return b;
+}
+// TODO
```

Comment:
What's left TODO here?

Reply:
It was only during the write up of the c code